cmd/modelcmd/base.go

```diff
@@ -438,6 +439,8 @@ func newAPIConnectionParams(
 	}
 	dialOpts := api.DefaultDialOpts()
 	dialOpts.BakeryClient = bakery
+	// we should not use default timeout - 10 minutes.
+	dialOpts.Timeout = 15 * time.Second
```

Comment:
The default should be changed in api.DefaultDialOpts()
10 minutes seems too long but was put there to allow controllers that are upgrading etc to finish.
Maybe a timeout of say 1 minute is better